### ISSUE TITLE
[QUESTION] how to open todotxt?

### ISSUE THREAD

**Issue by wsw70** [mvgrimes/obsidian-todotxt-plugin#26]
(sorry for asking a question as an Issue but I did not see any Discussions available)

I installed the plugin as described but I must be missing something: how to actually start the plugin? 😶

**Comment by mvgrimes** [comment]
In the Obsidian config menu, you should see "Community Plugins". If TodoTxt was installed correctly, you should see an entry for it:

![image](https://github.com/mvgrimes/obsidian-todotxt-plugin/assets/92431/d3c698c5-1350-49a6-aeeb-bf349418d09a)

 You should be able to toggle the plugin on/off there.

**Comment by wsw70** [comment]
> You should be able to toggle the plugin on/off there.

Thank you and sorry for not having been clear - I meant how to use it in Obsidian? (it is enabled)

Sorry if this is obvious but I have been going back and forth through the README and did not understand how to actually make use of your plugin.

**Comment by mvgrimes** [comment]
If you already have a todotxt formatted TODO file, move it into your vault. It can have any name, but should have the `.todotxt` extension.

If you don't already have TODO file, I think you can create a new note in Obsidian and name it anything you like. It will have the `.md` extension by default. You'll need to change it to `.todotxt`. Right-click the note in the list on the left and select "Reveal in Finder/Explorer" and you can hopefully rename it there.

**Comment by wsw70** [comment]
@mvgrimes I must be seriously missing something - sorry for following up.

I did have that file (`TODO.todotxt`) in my vault - this was described in the README. I am not sure how to state it, but now what? 🙂

**Comment by vcg3rd** [comment]
I think I understand what the OP is trying to say because I am having the same problem, and I have also been back and forth to the documentation trying to figure it out.  I.m using Obsidian-1.4.16.AppImage on Debian Bookworm.  After install, this plugin does not show a settings gear like in the screenshot above.  

![obsidian_todotxt](https://github.com/mvgrimes/obsidian-todotxt-plugin/assets/3229592/db8f94de-501b-4fdd-9752-e0a62aeebff2)  

I created a new note and gave it the .todotxt extension, then copied my todo.txt file into it, but the note behaves like a typical markdown note (including the annoying double space at the end of a line for /n).  There are no features of this plugin available.  

![obsidian_todotxt2](https://github.com/mvgrimes/obsidian-todotxt-plugin/assets/3229592/9c0ef3e8-dc52-467f-9016-c09966a73f01)

**Comment by mvgrimes** [comment]
@vcg3rd, can you also confirm that the `todo.todotxt` file doesn't have a `.md` extension? Obsidian typically hides the extension when displaying the filename. The best way is through a file browser or using `ls` from the command line (something outside of Obsidian).

Assuming the file extension is correct, I'm going guess there is a problem initializing the plugin on Debian or in an AppImage. I have not testing it on Linux yet, but I'll give it a try when time permits. In the meantime, if you can open the development tools/console log in Obsidian, you might find an error message which would be helpful. Depending on which browser Obsidian is using under the hood, I think the dev tools can be opened by pressing Ctl-Alt-J or Ctl-Alt-I while in Obsidian.

@wsw70, are you also seeing the plugin listed under "Community Plugins" but *without* the gear symbol?

**Comment by vcg3rd** [comment]
Yes.  Obsidian had added .md after the file name.  Removing it the note in the note list now shows as todo.todotxt `TODOTXT`.

I had a bit of a struggle to access Developer Tools.  No keyboard shortcut worked, nor was one listed as a possibility, and search and command palette both had zero results.  I finally changed the setting in Appearance | Windows Frame Style from nothing to Obsidian, and I was able to access it in the View menu.

I don't know any JS, but the only "warning" was:

```
app.js:1 Uncaught (in promise) Error: Attempting to register an existing file extension "todotxt"
    at t.registerExtensions (app.js:1:1986648)'
    at t.registerExtensions (app.js:1:2237808)'
    at TodotxtPlugin.onload (plugin:todotxt:28768:10)
```

I hope that helps.  At any rate, thanks a bunch.  It's your plugin, boss; I'm just passing through.  Take your time.  I am quite patient with OSS, and I am glad to help test and report, but that's the extent of my ability.

**Comment by mvgrimes** [comment]
That seems to imply that there is more than one plugin that has registered the `todotxt` file extension. Do you have another TODO plugin installed? Could this plugin have been installed twice somehow (ie, both manually and through the community plugin install process)?

**Comment by vcg3rd** [comment]
Sure enough.  I had tried out todotxt codeblocks, but I didn't like it.  After disabling it Obsidian (this plugin probably) threw up an error message about expected + @ [lots of other jazz] but found end?  I dunno.  I vanished quickly.  It may have not been necessary, but out of frustration I just deleted the file, recopied my todo.txt file in again, deleted the MD extension again, and BAM!  Magic!  Thanks a billion!!!

Having done that twice, it occurred to me I never needed to have created a note.  I don't want my todo.txt file unavailable to the command line or other tools including mobile.  So, I deleted it again then created a symlink named todo.todotxt to my todo.txt file in my Obsidian vault and this plugin still works, and I still have the one file!
